### PR TITLE
fix: Update value of expression column names key in the JSON object with new column names

### LINKED ISSUE #3: bug: ArgumentException thrown when serializing query to JSON containing DruidExpression with additional fields set
An `ArgumentException` is being thrown in the [SectionFactoryJsonMapper.cs](https://github.com/oskar11120/Apache.Druid.Querying/blob/master/Apache.Druid.Querying/Internal/Sections/SectionFactoryJsonMapper.cs#L44) when serializing a query to JSON containing `combine`/`compare`/`finalize` fields in addition to the `fold` field in the `DruidExpression`.

The exception is thrown because of adding `fields` property to the `JsonObject`, which has already been added.

Sample code:
```c#

private record Activity(
        [property: DataSourceTimeColumn] DateTimeOffset Timestamp,
        int Duration,
        int DomainID,
        int UserID);

private record Dimensions(int DomainID);

private record Aggregations(List<long> UserIds, int Duration);

var query = new Query<Activity>
            .GroupBy<Dimensions>
            .WithNoVirtualColumns
            .WithAggregations<Aggregations>()
            .Dimensions(type => new Dimensions(type.Default(activity => activity.DomainID)))
            .Aggregations(type => new Aggregations(
                type.Expression<List<long>, string>(
                    "ARRAY<LONG>[]",
                    "acc",
                    data => $"array_set_add(acc, {data.UserID})",
                    data => $"array_set_add_all(acc, UserIds)",
                    null,
                    null,
                    data => "ARRAY<LONG>[]",
                    true,
                    true,
                    false,
                    1024 * 10
                    ),
                type.Sum(activity => activity.Duration))
            )
            .Interval(new Interval(DateTimeOffset.UtcNow.AddYears(-1), DateTimeOffset.Now))
            .Granularity(SimpleGranularity.All);
        var json = query.MapToJson();
```

### PR DESCRIPTION
**Pull request by tomasko126**
This resolves https://github.com/oskar11120/Apache.Druid.Querying/issues/3 by checking for expression column names key in the JSON object before adding it. In case such key exists in the JSON object, its value gets updated with new column names.

### CODE REVIEW

File: Apache.Druid.Querying/Internal/Sections/SectionFactoryJsonMapper.cs [hunk 1]
```diff
@@ -40,8 +40,23 @@ void MapCallParam(ElementFactoryCall.Parameter.Any param, string? callResultMemb
...
+                        }
+                        else
+                        {
+                            result.TryGetPropertyValue(existing, out JsonNode? node);
+                            
+                            string[] fields = node.Deserialize<string[]>()!;
+                            
+                            result.Remove(existing);
+                            result.Add(existing, JsonSerializer.SerializeToNode(fields.Concat(columnNames), context.QuerySerializerOptions));
+                        }
```
Comment (oskar11120) [inline Apache.Druid.Querying/Internal/Sections/SectionFactoryJsonMapper.cs]: Thanks for the pr :)

Nickpick, but we can use `JsonNode.Remove` to both check existance of, get value of and remove the json property resulting in a bit more concise code.

Plus, I expect if multiple druid expressions use the same column, we don't want to include it multiple times, so we should do a `Distinct` on the column names.
  
```suggestion
                        if (result.Remove(existing, out var node))
                        {
                            var existingColumnNames = node.Deserialize<IEnumerable<string>>()!;
                            columnNames = existingColumnNames.Concat(columnNames).Distinct().ToArray();
                        }

                        result.Add(existing, JsonSerializer.SerializeToNode(columnNames, context.QuerySerializerOptions));
                        }
```

File: Apache.Druid.Querying.Tests.Unit/DataSerializerShould_ApplyTo.cs [hunk 2]
```diff
@@ -70,6 +71,47 @@ public void DimensionMetric()
...
+        [property: DataSourceTimeColumn] DateTimeOffset Timestamp,
+        int Duration,
+        int DomainID,
+        int UserID
+    );
+    private sealed record ActivityDimensions(int DomainID);
+    private sealed record ActivityAggregations(List<long> UserIds, int Duration);
+    
+    [Test]
+    public void ExpressionAggregationWithCombine()
```
Comment (oskar11120) [inline Apache.Druid.Querying.Tests.Unit/DataSerializerShould_ApplyTo.cs]: Let's move this test into `QueryShould.MapToRightJson` or into a completely new file. The tests in `DataSerializerShould_ApplyTo` are meant for making sure `DataSerializer` (serializer for "data" parts of the query, applying  "data conventions" e.g. implicitly converting booleans to numbers) applies to right parts of the query.

File: Apache.Druid.Querying/Internal/Sections/SectionFactoryJsonMapper.cs (hunk 1 above)
Comment (tomasko126) [reply Apache.Druid.Querying/Internal/Sections/SectionFactoryJsonMapper.cs]: Thanks for the suggestion to include calling `Distinct` method on column names! The suggestion looks definitely cleaner :)

File: Apache.Druid.Querying.Tests.Unit/DataSerializerShould_ApplyTo.cs (hunk 2 above)
Comment (tomasko126) [reply Apache.Druid.Querying.Tests.Unit/DataSerializerShould_ApplyTo.cs]: Thank you for the explanation, I have moved the unit test to the `QueryShould.MapToRightJson` 👌

**Review by oskar11120 — APPROVED** [review]

**Pull request merged**